src/components/Timeline.jsx

```diff
@@ -1,34 +1,64 @@
+/* eslint-disable no-undef */
 import React, {useContext, useState, useEffect} from 'react';
 import Sidebar from './Sidebar';
 import styled from 'styled-components';
 import {AuthContext} from '../contexts/AuthContext';
+import {LocationContext} from '../contexts/LocationContext';
 import firebase from '../plugins/firebase';
 import PostForm from './PostForm';
 import ShoutComponent from './ShoutComponent';
 
 const Timeline = () => {
-  const {auth, logout} = useContext(AuthContext);
+  const {logout} = useContext(AuthContext);
+  const {setLocation} = useContext(LocationContext);
   const [shoutData, setShoutData] = useState([]);
+  const [shoutSubmit, setShoutSubmit] = useState(false);
 
   useEffect(() => {
-    let shouts = [];
+    setLocation('/timeline');
+
+    const users = [];
```

Comment:
I feel like this is much easier to read.
And getting all of the users' data can be very big cost I think.
```javascript
useEffect(() => {
  setLocation('/timeline');
  const getShouts = async () => {
    const shoutsQuerySnapshot = await firebase
      .firestore()
      .collection('shouts')
      .orderBy('createdAt', 'desc')
      .limit(20)
      .get();
    const shouts = [];
    await asyncForEach(shoutsQuerySnapshot.docs, async doc => {
      const shout = doc.data();
      const userQuerySnapshot = await db
        .collection('users')
        .doc(shout.userId)
        .get();
      if (userQuerySnapshot.exists) {
        shouts.push({
          data: shout,
          userData: userQuerySnapshot.data(),
          id: doc.id,
        });
      }
    });
    setShoutData(shouts);
  };
  getShouts();
}, []);
```